### PR TITLE
Convert the Pages Module and some part of the Canvas Module to ts

### PR DESCRIPTION
**Pull request by xQwexx**

### CODE REVIEW

File: src/abstract/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export { default as Model } from './Model';
-export { default as Collection } from './Collection';
-export { default as View } from './View';
-export { default as Module } from './moduleLegacy';
+export { default as Model } from "./Model";
+export { default as Collection } from "./Collection";
+export { default as View } from "./View";
+export { default as Module } from "./Module";
```
Comment (artf) [inline src/abstract/index.ts:4]: Out of curiosity, any reason for why switching from single quotes to double quotes? 
As I don't see any change on the eslint config file, are you doing that change manually?

**Review by artf — CHANGES_REQUESTED** [review]
Nice 👏

File: src/abstract/index.ts (hunk 1 above)
Comment (xQwexx) [reply src/abstract/index.ts:4]: So I think my editor do this I am not sure where its gets the config

File: src/canvas/model/Canvas.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { Model } from "../../common";
+import Backbone from "backbone";
+import { evPageSelect } from "../../pages";
+import Frames from "./Frames";
+import EditorModel from "../../editor/model/Editor";
+import Page from "../../pages/model/Page";
+
+export default class Canvas extends Backbone.Model {
```
[the remark this answers is not in the document]
Comment (xQwexx) [reply src/canvas/model/Canvas.ts:8]: I going to make a separate pr for that I already on the process of making it just I have 6 test cases which are failing and I am unsuccessful to debug it so far.

File: src/pages/index.ts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  model: Model;
+  /**
+   * Initialize module
+   * @param {Object} config Configurations
+   * @private
+   */
+  constructor(em: EditorModel) {
+    super(em, "PageManager", new Pages([]), events);
+    bindAll(this, "_onPageChange");
+    this.getAll;
```
[the remark this answers is not in the document]
Comment (xQwexx) [reply src/pages/index.ts]: Yeah I deleted it

File: src/abstract/index.ts (hunk 1 above)
Comment (artf) [reply src/abstract/index.ts:4]: 😬  are you able to fix it as it brings a bit of inconsistency?

**Pull request merged**